### ISSUE TITLE
page not found

### ISSUE THREAD

**Issue by s4msung** [pug-php/pug#40]
When template was not found, it silently render "page not found" instead.
I just don't like this approach.

contra:
- localization
- issues with templates not found are possibly overlooked

+1 for (maybe optional) raising an exception if a template was not found

pro:
- testsuite captures uncatched exceptions

**Comment by kylekatarnls** [comment]
It already exists:
https://github.com/pug-php/pug/blob/master/tests/features/settings.php#L368

``` php
$pug = new Pug(array(
  'notFound' => false,
));
$pug->render('include does-not-exists');
```

This will throw an InvalidArgumentException with the code 22 and the message:
`The included file 'does-not-exists' does not exists.`

Please let me know if you get trouble with this option.

Explaination: the notFound option allow you to customize the rendering of missing templates. You can pass a pug code that will replace the missing include or `false` to throw the exception above.

**Comment by s4msung** [comment]
Ah, I've overlooked this options.

Now I got another problem:

`Paser::getTemplatePath` returns nothing when file does not exists, hence the exception thrown in `Parser::getTemplateContents` is kinda useless for debugging purposes

**Comment by kylekatarnls** [comment]
When `getTemplatePath` does not return a path, `getTempalteContents` will receive null as `$path`, so in this case the exception is thrown.

Please give me an example of what you want to debug and what information you need to get.

**Comment by s4msung** [comment]
An actual usefull exception message instead of `"The included file '' does not exists."`

**Comment by kylekatarnls** [comment]
Thanks! I'll fix it right now.

**Comment by kylekatarnls** [comment]
Thanks again, fixed in 2.0.3.
